src/etl/impl/SubscriptionSource.cpp

```diff
@@ -312,7 +319,11 @@ SubscriptionSource::logError(util::requests::RequestError const& error) const
 void
 SubscriptionSource::setLastMessageTime()
 {
-    lastMessageTime_.lock().get() = std::chrono::steady_clock::now();
+    lastMessageTimeSecondsSinceEpoch_.get().set(
+        std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count()
+    );
+    auto lock = lastMessageTime_.lock();
```

Comment:
I think previously it is locked already. But does it really need to be locked? The call path seems be in a strand.

Reply:
Getter for `lastMessageTime_` is called from another thread.